src/dex-ui/pages/CreateProposal/ProposalForms/ContractUpgradeProposalForm.tsx

```diff
@@ -1,16 +1,200 @@
+            />
+            <FormErrorMessage>{errors.title && errors.title.message}</FormErrorMessage>
+          </FormControl>
+          <FormControl>
+            <TextEditor id="description" placeholder="Description" {...register("description")} />
+          </FormControl>
+          <FormControl>
+            <Input
+              variant="form-input"
+              id="linkToDiscussion"
+              placeholder="Link to Discussion (optional)"
+              {...register("linkToDiscussion")}
+            />
+          </FormControl>
+          <FormControl>
+            <Flex direction="column" justifyContent="right" gap="0.5rem">
+              <FileUploader
+                id="abiFile"
+                title="Drag and Drop your ABI file"
+                subTitle="or click to browse your files"
+                onHoverTitle="Drop your ABI file here ..."
+                onFileRead={onFileReadAction}
+              />
+              {
+                fileName.length !== 0 && (
+                  <Text textStyle="b3" color={Color.Blue_02}>
+                    {" "}
+                    {fileName}
+                  </Text>
+                ) // TODO: Place for the file name to show
+              }
+            </Flex>
+          </FormControl>
+          <FormControl isInvalid={Boolean(errors.newImplementationAddress)}>
+            <Input
+              disabled
+              variant="form-input"
+              id="newImplementationAddress"
+              placeholder="New Implementation Address"
+              {...register("newImplementationAddress", {
+                required: { value: true, message: "Address is required." },
+              })}
+            />
+            <FormErrorMessage>
+              {errors.newImplementationAddress && errors.newImplementationAddress.message}
+            </FormErrorMessage>
+          </FormControl>
+          <FormControl isInvalid={Boolean(errors.proxyAddress)}>
+            <Input
+              variant="form-input"
+              id="proxyAddress"
+              placeholder="Proxy Address"
+              {...register("proxyAddress", {
+                required: { value: true, message: "Proxy Address is required." },
+              })}
+            />
+            <FormErrorMessage>{errors.proxyAddress && errors.proxyAddress.message}</FormErrorMessage>
+          </FormControl>
+          <Spacer padding="0.5rem" />
+          <Flex direction="row" justifyContent="right" gap="0.5rem">
+            <Button variant="secondary" padding="10px 27px" height="40px" onClick={handleCancelClick}>
+              Cancel
+            </Button>
+            <Button
+              type="submit"
+              variant="primary"
+              padding="10px 27px"
+              height="40px"
+              isLoading={isSubmitting}
+              alignSelf="end"
+            >
+              Publish
+            </Button>
+          </Flex>
+        </Flex>
+      </form>
+      <LoadingDialog
```

Comment:
In the future, you can use a single `LoadingDialog` component for loading and a single `LoadingDialog` component for error states. Take a look at `ProposalDetails` component and `useProposalDetails` hook to see how we are currently doing this. We can enhance the LoadingDialog component in the future to directly accept getter functions for **isOpen** and **message** values.

Reply:
Done